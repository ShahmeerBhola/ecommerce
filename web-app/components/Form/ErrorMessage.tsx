import React, { FC } from 'react';

type Props = {
    error: Forms.FormikError;
};

const ErrorMessage: FC<Props> = ({ error }) => {
    return (
        <div className="absolute text-red-100 text-xs">
            <span>{error}</span>
        </div>
    );
};

export default ErrorMessage;
