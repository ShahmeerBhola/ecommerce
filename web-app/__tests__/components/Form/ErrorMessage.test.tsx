import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from '../../../components/Form/ErrorMessage';

describe('ErrorMessage component', () => {
  test('it renders nothing if error is null and touched is false', () => {
    const tree = renderer.create(<ErrorMessage error={null} touched={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders nothing if error is non-null and touched is false', () => {
    const tree = renderer.create(<ErrorMessage error="This is not valid!" touched={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders nothing if error is null and touched is true', () => {
    const tree = renderer.create(<ErrorMessage error={null} touched />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders message if error is non-null and touched is true', () => {
    const tree = renderer.create(<ErrorMessage touched error="This is not valid!" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
