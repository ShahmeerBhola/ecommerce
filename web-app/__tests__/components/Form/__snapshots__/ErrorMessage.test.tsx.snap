// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component it renders message if error is non-null and touched is true 1`] = `
<p
  className="text-red-100 text-xs pt-1"
>
  This is not valid!
</p>
`;

exports[`ErrorMessage component it renders nothing if error is non-null and touched is false 1`] = `null`;

exports[`ErrorMessage component it renders nothing if error is null and touched is false 1`] = `null`;

exports[`ErrorMessage component it renders nothing if error is null and touched is true 1`] = `null`;
